Reference region cleanup

fnothaft opened the pull request:
Resolves #1285 and #1286. The first two commits need to be squashed together.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1637/
Test PASSed.

fnothaft commented:
Ping @heuermh @jpdna for review.

heuermh commented on adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala:
```diff
@@ -220,17 +255,45 @@ object ReferenceRegion {
   }
 
   /**
+   * Builds a reference region for a feature without strand information.
```
`without strand information.` &rarr; `with strand set.`

heuermh commented on adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala:
```diff
@@ -220,17 +255,45 @@ object ReferenceRegion {
+  def stranded(feature: Feature): ReferenceRegion = {
+    checkFeature(feature)
+    require(feature.getStrand != null,
+      "Strand is not defined in feature %s.".format(feature))
```
Add @throws doc for this

heuermh commented on adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala:
```diff
@@ -148,7 +148,9 @@ object ReferenceRegion {
    */
   def apply(genotype: Genotype): ReferenceRegion = {
-    ReferenceRegion(genotype.getVariant)
+    ReferenceRegion(genotype.getContigName,
```
A few general questions (not about this line specifically):

Rename `orientation` to `strand`?
Add strand parameters to other methods (e.g. `fromStart`, `toEnd`, `all`)?
Set strand when copying from one `ReferenceRegion` to another (i.e. line 234)?

fnothaft commented:
Thanks for the comments @heuermh. I've rebased and pushed two commits addressing your comments. When this looks good to you, I'll squash commits 2–4 down into a single commit for merge.

fnothaft commented:
Squashed down.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1667/
Test PASSed.

heuermh commented:
Thank you, @fnothaft

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1668/
Test PASSed.
